sync/src/sync.rs

```diff
@@ -233,7 +233,7 @@ impl SyncService {
                 storage.get_block_info(current_block_id)?.ok_or_else(|| {
                     format_err!("Can not find block info by id: {}", current_block_id)
                 })?;
-            let dag_fork_height = chain_service.dag_fork_height().await?;
+            let dag_fork_height = Some(config.net().genesis_config().dag_effective_height);
```

Comment:
For safety, we need get it from statedb,  the pre-set value is only used when executing genesis block technically.

Reply:
it means this value can be changed through somewhere, such as framework?

Reply:
> it means this value can be changed through somewhere, such as framework?

Yes. DAO is one way to reset it.

Reply:
Why do we still need dag_fork_height ?

Reply:
Because the code maintains synchronization logic of the sing chain and dag and it is flexible.

Reply:
I remember that sanlee42 suggests that we use some cfg macros @sanlee42 .